### PR TITLE
Feature/gin binding and validation

### PR DESCRIPTION
**Pull request by Mikelle**
Closes https://github.com/hermeznetwork/hermez-node/issues/837
### What does this PR does?

This PR moves logic from custom implementation for validation and binding to gin lib

### How to test?

Run api package tests

## Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Include tests
- [x] Respect code style and lint
- [ ] Update swagger file (if needed)
- [ ] Update documentation (*.md) (if needed)

### CODE REVIEW

File: api/parsers/coordinator.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	"github.com/hermeznetwork/hermez-node/common"
+	"github.com/hermeznetwork/hermez-node/db/historydb"
+	"github.com/hermeznetwork/tracerr"
+)
+
+type coordinatorsFilters struct {
+	BidderAddr string `form:"bidderAddr"`
+	ForgerAddr string `form:"forgerAddr"`
+
+	Pagination
```
Comment (jeffprestes) [inline api/parsers/coordinator.go:14]: Is this correct ?

File: api/parsers/exits.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+import (
+	"github.com/gin-gonic/gin"
+	"github.com/hermeznetwork/hermez-node/common"
+	"github.com/hermeznetwork/hermez-node/db/historydb"
+	"github.com/hermeznetwork/tracerr"
+	"gopkg.in/go-playground/validator.v9"
+)
+
+type exitFilter struct {
```
Comment (jeffprestes) [inline api/parsers/exits.go:11]: Maybe would be a nice idea left those structs within the parsers public to allow a third-party library to reuse them.

File: api/parsers/fiat.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
+package parsers
+
+import (
+	"strings"
+
+	"github.com/gin-gonic/gin"
+)
+
+type currencyFilter struct {
```
Comment (jeffprestes) [inline api/parsers/fiat.go:9]: Maybe would be a nice idea left those structs within the parsers public to allow a third-party library to reuse them.

File: api/parsers/slots.go [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+// ParseSlotFilter func to parse slot filter from uri to the slot number
+func ParseSlotFilter(c *gin.Context) (*uint, error) {
+	var slotFilter slotFilter
+	if err := c.ShouldBindUri(&slotFilter); err != nil {
+		return nil, tracerr.Wrap(err)
+	}
+	return slotFilter.SlotNum, nil
+}
+
+type slotsFilters struct {
```
Comment (jeffprestes) [inline api/parsers/slots.go:23]: Maybe would be a nice idea left those structs within the parsers public to allow a third-party library to reuse them.

**Review by jeffprestes — CHANGES_REQUESTED** [review]
Maybe would be a nice idea left filter structs within the parsers package public to allow a third-party library to reuse them.

**Review by jeffprestes — APPROVED** [review]

**Review by arnaubennassar — APPROVED** [review]

**Pull request merged**